packages/foam-vscode/src/features/utility-commands.ts

```diff
@@ -16,16 +16,21 @@ export const OPEN_COMMAND = {
         return vscode.commands.executeCommand('vscode.open', toVsCodeUri(uri));
 
       case 'placeholder':
-        const newNote = await createNoteFromPlaceholder(uri);
-
-        if (isSome(newNote)) {
-          const title = uri.path.split('/').slice(-1);
-          const snippet = new vscode.SnippetString(
-            '# ${1:' + title + '}\n\n$0'
-          );
-          await focusNote(newNote, true);
-          await vscode.window.activeTextEditor.insertSnippet(snippet);
-        }
+        const title = uri.path.split('/').slice(-1)[0];
```

Comment:
can you walk me through this change?

have you embedded the `createNoteFromPlaceholder` function and using the template pattern at the same time?

Reply:
> can you walk me through this change?
@riccardoferretti 

* `OPEN_COMMAND` is the command used when other parts of Foam want to follow a placeholder Wikilink:
  * It calls `createNoteFromPlaceholder` to find the location for the new note, and create an empty file there.
  * If it could not determine where to create the file, then it returns, doing nothing.
  * `title` is the contents of the wikilink, which it writes into the new note as a snippet placeholder

I have changed `OPEN_COMMAND` so that it uses the new `createNoteForPlaceholderWikilink` to maintain the same behaviour:
  * `basedir` uses the same logic from `createNoteFromPlaceholder` to determine where to open the new note.
    * If the default template specifies a `filepath` metadata attribute, then that will be used instead
    * If it cannot be determined, `OPEN_COMMAND` still returns without doing anything.
      * In practice, it doesn't seem possible for this to happen. If there is no active note, what placeholder wikilink are you clicking on??
      * Besides, Foam doesn't (yet) work at all if you don't have a workspace open.
        * I really want to fix all the `vscode.workspace.workspaceFolders[0]` instances, but that is not something for this PR to fix.
  * `title` is the contents of the wikilink, which is used as the value for `$FOAM_TITLE`, as per #566
  * `wikilinkDefaultTemplateText` matches the current behaviour's file contents, but of course will only be used if the `new-note.md` template doesn't exist.

> have you embedded the `createNoteFromPlaceholder` function

The default filepath location logic is now embedded in `OPEN_COMMAND`. 
The file creation logic is now done by the `create-from-template` code.
Now that `createNoteFromPlaceholder`'s functionality has been subsumed, it has been removed.

Reply:
Thanks for this.
I agree with your sentiment about `workspaceFolders`. We probably need to consolidate the path resolving logic for notes/placeholders/... in one place (we touched on this before). Something that would consistently apply the proper basedir prioritization, resolve relative paths, append the extension if necessary, ... once that's in place we could add some configuration to it, e.g. note extension, slugification/casification/...

But not for this PR ;)